test/distributed/test_c10d_nccl.py

```diff
@@ -420,6 +420,47 @@ def allgather_base(output_t, input_t):
             # fails the check because the dtype is different
             allgather_base(output_t, tensor)
 
+    @requires_nccl()
+    def test_gather_ops(self):
+        store = c10d.FileStore(self.file.name, self.world_size)
+        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+
+        def gather(output_t, input_t, rootRank):
+            opts = c10d.GatherOptions()
+            opts.rootRank = rootRank
+            if rootRank == self.rank:
+                work = pg.gather(output_t, input_t, opts)
+            else:
+                work = pg.gather([], input_t, opts)
+            work.wait()
+
+        # init input
+        tensors = []
+        for i in range(self.num_gpus):
+            tensors.append(torch.tensor([self.rank]).cuda(i))
```

Comment:
We should also test a case where the input tensor is of different length on each rank.

Reply:
I don't see a test where we use different lengths of tensors?